modules/EnsEMBL/Web/SpeciesDefs.pm

```diff
@@ -892,124 +907,139 @@ sub _parse {
+          is_leaf         => 'true'
+        };
 
-          if (!$node->{taxa}) {
-            push @{$other_species[0]->{child_nodes}}, $child;
-          }
-          else {
-            my %taxa = map {$_ => 1} @{ $node->{taxa} };
-            my @matched_groups = grep { $taxa{$_} } @$taxonomy;
-            if ($#matched_groups >= 0) {
-              if ($node->{child_nodes}) {
-                my $cnode_match = {};
-                foreach my $cnode ( @{$node->{child_nodes}}) {
-                  my @match = grep { /$matched_groups[0]/ }  @{$cnode->{taxa}};
-                  if ($#match >=0 ) {
-                    $cnode_match = $cnode;
-                    last;
-                  }
-                }
+        if ($strain_group && $strain_name !~ /reference/) {
+          $child->{type} = $group_name . ' ' . $tree->{$url}{'STRAIN_TYPE'}. 's';
+        }
+        elsif($strain_group && $strain_name =~ /reference/) {
+          # Create display name for Reference species
+          my $ref_name = $tree->{$url}{'SPECIES_DISPLAY_NAME'} . ' '. $strain_name;
+          $child->{display_name} = $ref_name;
+        }
 
-                if (keys %$cnode_match) {
-                  if (!$cnode_match->{child_nodes}) {
-                    $cnode_match->{child_nodes} = [];
-                  }
-                  push @{$cnode_match->{child_nodes}}, $child;
+        if (!$node->{taxa}) {
+          push @{$other_species[0]->{child_nodes}}, $child;
+        }
+        else {
+          my %taxa = map {$_ => 1} @{ $node->{taxa} };
+          my @matched_groups = grep { $taxa{$_} } @$taxonomy;
+          if ($#matched_groups >= 0) {
+            if ($node->{child_nodes}) {
+              my $cnode_match = {};
+              foreach my $cnode ( @{$node->{child_nodes}}) {
+                my @match = grep { /$matched_groups[0]/ }  @{$cnode->{taxa}};
+                if ($#match >=0 ) {
+                  $cnode_match = $cnode;
                   last;
                 }
-                else {
-                  if (!$node->{child_nodes}) {
-                    $node->{child_nodes} = [];
-                  }
-                  push @{$node->{child_nodes}}, $child;
-                  last;
+              }
+
+              if (keys %$cnode_match) {
+                if (!$cnode_match->{child_nodes}) {
+                  $cnode_match->{child_nodes} = [];
                 }
+                push @{$cnode_match->{child_nodes}}, $child;
+                last;
               }
               else {
-                $node->{child_nodes} = [];
+                if (!$node->{child_nodes}) {
+                  $node->{child_nodes} = [];
+                }
                 push @{$node->{child_nodes}}, $child;
                 last;
               }
             }
+            else {
+              $node->{child_nodes} = [];
+              push @{$node->{child_nodes}}, $child;
+              last;
+            }
           }
         }
       }
     }
-    else {
```

Comment:
I can see this has been removed. so probably an if statement is gone from there. Can you check if thats intended? seems like a valid error to me or maybe not.

Reply:
See above.